sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -201,9 +201,11 @@ def get_postprocess_fns(self) -> Iterable[Callable[[Any], Any]]:
 
   def set_environment_vars(self):
     """Sets environment variables using a dictionary provided via kwargs.
-    Keys are the env variable name, and values are the env variable value."""
-    for env_variable, env_value in self._env_vars.items():
-      os.environ[env_variable] = env_value
+    Keys are the env variable name, and values are the env variable value.
+    Child ModelHandler classes should set _env_vars via kwargs in __init__"""
+    if '_env_vars' in self.__dict__:
```

Comment:
Accessing self.__dict__ directly is a bit hacky, and doesn't always work. Intead, maybe do

```
for env_variable, env_value in getattr(self, '_env_vars', {}).items():
```

or guard with `hasattr(self, '_env_vars')`

Reply:
I changed it to:
 env_vars = getattr(self, '_env_vars', {})
for env_variable, env_value in env_vars:

otherwise I got errors about getattr not being iterable.